### PR TITLE
Feature/5 user interface for instance creation

### PR DESCRIPTION
**Pull request by Munin33**

### CODE REVIEW

File: src/views/CreateInstance.vue [hunk 1]
```diff
@@ -0,0 +1,157 @@
+<template>
+  <main v-if="formState === 0" class="create-Instance">
+    <ViewHeader
+      :title="`Choose a resource type for the new instance`"
```
Comment (MaximilianV) [inline src/views/CreateInstance.vue]: This does not need to be a template string, and `:` can be removed, as you have no variable binding here.

Comment (MaximilianV) [reply src/views/CreateInstance.vue]: Please use consistent upper/lower case for `instance`. In the next step, instance is written in upper case

File: src/views/CreateInstance.vue [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      :backLink="{ link: '/resources' }"
+    />
+    <ListSection :list="nonAbstractResourceTypesList"/>
+  </main>
+
+  <main v-else-if="formState === 1" class="create-Instance">
```
Comment (MaximilianV) [inline src/views/CreateInstance.vue]: classes should always be written in lower case. Is it even needed for styling?

File: src/views/CreateInstance.vue [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    TypeSensitiveInput,
+    Input,
+    Toggle,
+    Button,
+    Select,
+  },
+})
+export default class Types extends Mixins(Translate) {
+  // region public static methods
+  public static emptyResourceType(): ResourceType {
```
Comment (MaximilianV) [inline src/views/CreateInstance.vue]: Why not use the `ResourceTypeNullObject` here?
https://github.com/bptlab/rembrandt-frontend/blob/df1ea54fde75620c5680b8ec4db652f39570fcc7/src/apis/rembrandt/lib/ResourceType.ts#L37

File: src/views/CreateInstance.vue [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    Li,
+    ViewHeader,
+    TypeSensitiveInput,
+    Input,
+    Toggle,
+    Button,
+    Select,
+  },
+})
+export default class Types extends Mixins(Translate) {
```
Comment (MaximilianV) [inline src/views/CreateInstance.vue]: Why is this class named `Types`?

File: src/views/CreateInstance.vue [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+  // region private members
+  // endregion
+
+  // region constructor
+  constructor() {
+    super();
+    this.newResourceInstance = ResourceInstanceNullObject;
+    this.newResourceInstance.attributes = [];
+    this.resourceType = Types.emptyResourceType();
```
Comment (MaximilianV) [inline src/views/CreateInstance.vue]: `Types.` due to misnamed class?

File: src/views/CreateInstance.vue [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      this.newResourceInstance.attributes = [];
+    }
+  }
+
+  public nextStep(): void {
+    this.formState++;
+    if (this.formState === 1) {
+      this.newResourceInstance.resourceType = this.resourceType;
+      this.resourceType.attributes.forEach( (attribute) => {
+      this.newResourceInstance.attributes.push({name: attribute.name, value: ''});
```
Comment (MaximilianV) [inline src/views/CreateInstance.vue]: Wrong indentation!

File: src/views/Resources.vue [hunk 7]
```diff
@@ -109,6 +114,38 @@ export default class Resources extends Vue {
 <style lang="less">
 @import "../colors.less";
 
+.create-instance-button {
```
Comment (MaximilianV) [inline src/views/Resources.vue]: Does it somehow differ from the button at types view?
If it's the same style, we need to extract it to avoid duplication

**Review by MaximilianV — CHANGES_REQUESTED** [review]

File: src/views/CreateInstance.vue (hunk 1 above)
Comment (Munin33) [reply src/views/CreateInstance.vue]: done

File: src/views/CreateInstance.vue (hunk 3 above)
Comment (Munin33) [reply src/views/CreateInstance.vue]: changed it

File: src/views/CreateInstance.vue (hunk 4 above)
Comment (Munin33) [reply src/views/CreateInstance.vue]: changed it to Instances

File: src/views/CreateInstance.vue (hunk 5 above)
Comment (Munin33) [reply src/views/CreateInstance.vue]: jep, but i changed it to the resourceTypeNullObject anyway

File: src/views/CreateInstance.vue (hunk 6 above)
Comment (Munin33) [reply src/views/CreateInstance.vue]: corrected it

File: src/views/Resources.vue (hunk 7 above)
Comment (Munin33) [reply src/views/Resources.vue]: I extracted it to the AddButton component

**Review by MaximilianV — DISMISSED** [review]
🚀 nice!

**Review by friedow — APPROVED** [review]

**Pull request merged**